.pipelines/cni/singletenancy/linux-cniv2-template.yaml

```diff
@@ -0,0 +1,179 @@
+  os: linux
+  arch: ""
+  testDropgz: true
+
+# + Should be able to scale up/down the pods successfully certain number of times.
+# + Node reboot scenarios should be covered.
+# + The CNS state should be validated with that of CNI state.
+# + Pods should have ips assigned and connectivity/datapath test should be present.
+# + CNS restart and validates the state
+
+stages:
+  - stage: create_${{ parameters.name }}
+    variables:
+      commitID: $[ stagedependencies.setup.env.outputs['SetEnvVars.commitID'] ]
+    pool:
+      name: $(BUILD_POOL_NAME_DEFAULT)
+    dependsOn:
+      - setup
+    displayName: "Create Cluster - ${{ parameters.clusterName }}"
+    jobs:
+      - job: create_aks_cluster_with_${{ parameters.name }}
+        steps:
+          - template: ../load-test-templates/create-cluster-template.yaml
+            parameters:
+              clusterType: ${{ parameters.clusterType }}
+              clusterName: ${{ parameters.clusterName }}-$(commitID)
+              nodeCount: ${{ parameters.nodeCount }}
+              vmSize: ${{ parameters.vmSize }}
+              region: $(LOCATION)
+
+  - stage: ${{ parameters.name }}
+    variables:
+      commitID: $[ stagedependencies.setup.env.outputs['SetEnvVars.commitID'] ]
+      dropgzVersion: $[ stagedependencies.setup.env.outputs['SetEnvVars.dropgzVersion'] ]
+      cnsVersion: $[ stagedependencies.setup.env.outputs['SetEnvVars.cnsVersion'] ]
+      npmVersion: $[ stagedependencies.setup.env.outputs['SetEnvVars.npmVersion'] ]
+    pool:
+      name: $(BUILD_POOL_NAME_DEFAULT)
+    dependsOn:
+      - create_${{ parameters.name }}
+      - publish
+      - setup
+    displayName: "CNIv2 Test - ${{ parameters.name }}"
+    jobs:
+      - job: integration
+        displayName: "Integration Test - ${{ parameters.name }}"
+        steps:
+          - ${{ if contains(parameters.clusterType, 'overlay') }}:
+            - task: AzureCLI@1
+              inputs:
+                azureSubscription: $(TEST_SUB_SERVICE_CONNECTION)
+                scriptLocation: "inlineScript"
+                scriptType: "bash"
+                addSpnToEnvironment: true
+                inlineScript: |
+                  echo "Start Integration Tests on Overlay Cluster"
+                  echo "deploy ip-masq-agent for overlay"
+                  make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}-$(commitID)
+                  kubectl apply -f test/integration/manifests/ip-masq-agent/ip-masq-agent.yaml --validate=false
+                  cd test/integration/manifests/ip-masq-agent/
+                  kubectl apply -f config-custom.yaml
+                  kubectl apply -f config-reconcile.yaml
+                  cd ../../../..
+                  kubectl get po -owide -A
+                  sudo -E env "PATH=$PATH" make test-integration CNS_VERSION=$(cnsVersion) CNI_DROPGZ_VERSION=$(dropgzVersion) INSTALL_CNS=true INSTALL_AZURE_CNI_OVERLAY=true TEST_DROPGZ=${{ parameters.testDropgz }}
```

Comment:
Since this is CNI load test, wouldn't it make sense to test the CNI binaries with the released CNS ?

Reply:
I agree, this slipped my mind when creating the variables for cns and npm. We need to ensure that there is visibility/consistency on which version should be used. The various `make *-version` do not output a version that can be consumed properly by either acnpublic or MCR. `v1.5.10-9-*` is the current output... My quick solution is setting it within the Variable Group, but that is not precise or widely used.